src/pages/workspaces/workspace/analysis/modals/ComputeModal.js

```diff
@@ -1756,16 +1756,20 @@ export const ComputeModalBase = ({
     ])
   }
 
-  const renderPersistentDiskSection = diskExists => {
+  const PersistentDiskSection = diskExists => {
     const gridStyle = { display: 'grid', gridGap: '1rem', alignItems: 'center', marginTop: '1rem' }
+    const diskSizeId = useUniqueId()
 
-    const renderPersistentDiskType = id => h(div, [
-      label({ htmlFor: id, style: computeStyles.label }, ['Disk Type']),
+    const PersistentDiskType = () => {
```

Comment:
If I am reading this right, PersistentDiskType is a React.FunctionalComponent being defined inside another functional component.  This is a pattern I've seen a few times in this codebase that we should discourage.  Render methods should be treated as performance-sensitive, and we should strive to have things like this defined at the module scope to not weigh down the render method itself.  It also can occlude things like usage of things from the outer component inside the inner defined one.  Cleaner to keep these things isolated as neighboring FCs.